src/components/NodeViewBase.tsx

```diff
@@ -0,0 +1,83 @@
+import React from 'react'
+import NodeView from './NodeView'
+import {
+  RudolfStore,
+  CustomDispatch,
+  updateJustification,
+} from '../RudolfReducer'
+import nodeviewJSS, { rowHeight } from '../styles/NodeView_styles'
+import { range } from '../util/helpers'
+import AutosizingInput from './AutosizingInput'
+
+type Props = {
+  currentState: RudolfStore
+  dispatch: CustomDispatch
+}
+
+const NodeViewBase = ({ currentState, dispatch }: Props) => {
+  const classes = nodeviewJSS()
+  const { nextRow, tree, justifications, feedback } = currentState
+  const rows = range(1, nextRow)
+  return (
+    <div
+      className={classes.NodeViewBaseContainer}
+      style={{
+        gridTemplateRows: `repeat(${nextRow - 1}, ${rowHeight})`,
+      }}
+    >
+      {rows.map((row) => {
+        return (
+          <div className={classes.RowNumber} key={row} style={{ gridRow: row }}>
+            {`${row}.`}
+          </div>
+        )
+      })}
+      <div
+        style={{
+          gridColumn: 'nodeView',
+          gridRow: `1 / span ${rows.length}`,
+        }}
+      >
+        <NodeView
+          node={tree}
+          dispatch={dispatch}
+          justifications={justifications}
+          feedbackMap={feedback.feedback}
+          currLastRow={nextRow - 1}
```

Comment:
This should be passed down as `nextRow` without the `- 1`, instead of creating a second related value.

The only time this gets used, you add the `+ 1` back anyways.

```
// in NodeView.tsx
gridTemplateRows: `repeat(${
            currLastRow - firstRow(node) + 1
          }, ${rowHeight})`,
```

Reply:
I did this when figuring out the solution, because I wanted to deal with the last row itself in NodeView and not next row. But css grid indices start at 1, so I had to add the one to offset that change. But anyways, change made.